File: guessing_game.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let randomNumber = Math.round(Math.random() * 100 + 1);
let attempts = 1;

function promptGuess() {
  rl.question("What's your guess?  ", (input) => {
    const guess = parseInt(input);

    if (input > 100 || isNaN(input)) {
      console.log("Out of range. Guess Again");
      promptGuess();
      return;
    }

    if (input == randomNumber) {
      console.log(`Correct! You guessed in ${attempts} attempts!`);
      rl.question("Would you like to play again? Y or N ", (answer) => {
        let real = answer.toUpperCase();
        switch (real) {
          case "Y":
            randomNumber = Math.round(Math.random() * 100 + 1);
            attempts = 1;
            promptGuess();
            break;
          case "N":
            rl.close();
            break;
          default:
            console.log("Invalid");
            rl.close();
        }
      });
      return;
    } else if (input > randomNumber) {
      console.log("Too High");
      attempts++;
      promptGuess();
    } else if (input < randomNumber) {
      console.log("Too Low");
      attempts++;
      promptGuess();
    }
  });
}
promptGuess();

File: guessing_game.f90
program hello
    implicit none
    real :: r 
    integer :: random_int
    integer :: guess_int
    integer :: attempts = 0;
    CHARACTER(len=3) :: guess
    CHARACTER(len=3) :: attempts_str

    call random_seed()
    call random_number(r)
    random_int = int(r * 100) + 1

    DO WHILE (guess_int /= random_int)
        attempts = attempts + 1
        WRITE(*,*) "Guess a number between 1-100: "
        READ(*,*) guess
        read (guess, '(I10)') guess_int

        IF (guess_int > random_int) THEN
            WRITE(*,*) "Too high"
        END IF

        IF (guess_int < random_int) THEN
            WRITE(*,*) "Too low"
        END IF
    END DO

    write(attempts_str, '(I0)') attempts

    WRITE(*,*) "You got it in " // attempts_str // "attempts"
end program hello
